lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -4,153 +4,634 @@
+
+  /// Returns `true` if the `Meta` key (on Mac keyboards, the `⌘ Command key`; on Windows keyboards, the Windows key (`⊞`))
+  /// was active when the key event was generated.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/metaKey>
   external bool get metaKey;
+
+  /// Returns `true` if the key is being held down such that it is automatically repeating.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat>
   external bool get repeat;
+
+  /// Returns `true` if the `Shift` key was active when the key event was generated.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/shiftKey>
   external bool get shiftKey;
+
+  /// Returns a [num] representing a system and implementation dependent numerical code identifying the unmodified
+  /// value of the pressed key.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode>
   external num get keyCode;
+
+  /// Returns a [num] representing the Unicode reference number of the key; this attribute is used only by the `keypress` event.
+  ///
+  /// For keys whose `char` attribute contains multiple characters, this is the Unicode value of the first character in
+  /// that attribute. In Firefox 26 this returns codes for printable characters.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode>
   external num get charCode;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [CompositionEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/CompositionEvent>
 @JS()
-@anonymous
 class SyntheticCompositionEvent extends SyntheticEvent {
+  /// [SyntheticCompositionEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticCompositionEvent` instead.
+  external factory SyntheticCompositionEvent._();
+
+  /// Returns the characters generated by the input method that raised the event; its varies depending on the type of event
+  /// that generated the `CompositionEvent` object.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/CompositionEvent/data>
   external String get data;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [FocusEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent>
 @JS()
-@anonymous
 class SyntheticFocusEvent extends SyntheticEvent {
+  /// [SyntheticFocusEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticFocusEvent` instead.
+  external factory SyntheticFocusEvent._();
+
+  /// An [EventTarget] representing a secondary target for this event.
+  ///
+  /// In some cases (such as when tabbing in or out a page), this property may be set to `null` for security reasons.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent/relatedTarget>
   external EventTarget get relatedTarget;
 }
 
+/// A [SyntheticEvent] wrapper that represents a form event.
```

Comment:
This doesn't exist in React.js and I feel like it should be removed. I started to remove it, but because it's used in several places it felt like that could be its own ticket. Thoughts?

Reply:
+1 to its own ticket